### PR TITLE
fix: Add Fixed Height and Scroll to Search Dialog Result

### LINKED ISSUE #77: Add Fixed Height and Scroll to Search Dialog Result
Currently, the search dialog result in the Jargons Editor does not have a fixed height or scroll functionality. This becomes an issue when the search result is extensive, causing it to overflow off the page and making it difficult for users to view all the results. To improve the user experience, we need to add a fixed height to the search dialog result and enable scrolling when the content exceeds the fixed height.

[screencast-localhost_4321-2024.06.07-22_40_17.webm](https://github.com/devjargons/jargons.dev/assets/25631971/c9d035e1-f21b-4812-8962-239f5c31b2e8)

#### Task
1. **Add Fixed Height**: Set a fixed height for the search dialog result div.
2. **Enable Scroll**: Add a vertical scroll bar to the search dialog result when the content exceeds the fixed height.

#### Implementation Steps
1. **Set Fixed Height**:
   - Determine an appropriate fixed height for the search dialog result.
   - Apply the fixed height using CSS to the search dialog result div.

2. **Enable Scroll**:
   - Add CSS properties to enable vertical scrolling for the search dialog result div when the content exceeds the fixed height.

#### Acceptance Criteria
- The search dialog result div should have a fixed height.
- When the content exceeds the fixed height, a vertical scroll bar should appear, allowing users to scroll through the content.

#### Additional Information
- Adding a fixed height and scroll functionality to the search dialog result will ensure that users can easily view and navigate through search results

### PR DESCRIPTION
**Pull request by jomefavourite**
### Description

This PR adds a fixed height to the search dialog result in order to enable scrolling when the content exceeds the screen/viewport height.

### Related Issue

Fixes #77

### Screenshots/Screencasts

image

### CODE REVIEW

File: src/components/islands/search.jsx [hunk 1]
```diff
@@ -222,7 +222,7 @@ function SearchResult({ result = [], cursor, searchTerm }) {
   const router = useRouter();
 
   return (
-    <div className="block w-full text-sm md:text-base">
+    <div className="block w-full text-sm md:text-base  max-h-[505px] overflow-y-scroll">
```
Comment (babblebey) [inline src/components/islands/search.jsx:225]: The maximum height (`max-h-**`) approach is great, exactly the thing that does it! But, I thought that the value of `505px` is a little rigid 🤔 

Take an instance where the viewport isn't even up to `505px` in size, the dialog would still enforce a `505px` hence shooting the div off the page... see screencast below 

[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2024.08.24-23_50_52.webm](https://github.com/user-attachments/assets/fbee595f-bcb7-4a12-90ef-85e559b7470a)

### Here's what I would do 😉 

- Transfer the max-width styling to the parent div of both the search form and search result div, setting it to `max-h-full` i.e. `max-height: 100%` making sure that the child respects the available width.

[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2024.08.24-23_58_26.webm](https://github.com/user-attachments/assets/7e2dba2e-3a52-474b-8a3b-e0cbde3321da)

**Review by babblebey — CHANGES_REQUESTED** [review]
Great stuff @jomefavourite,

Thank you for this one yea 😁, I've got some an observation, an improvement I think we can make 😉

**Comment by jomefavourite** [comment]
Hello @babblebey 
Apologies for the delay, I'll fix this today

**Comment by jomefavourite** [comment]
@babblebey I've made the fix

File: src/components/islands/search.jsx [hunk 2]
```diff
@@ -222,7 +222,7 @@ function SearchResult({ result = [], cursor, searchTerm }) {
   const router = useRouter();
 
   return (
-    <div className="block w-full text-sm md:text-base">
+    <div className="block w-full text-sm md:text-base overflow-y-scroll">
```
Comment (babblebey) [inline src/components/islands/search.jsx:225]:
```suggestion
    <div className="block w-full text-sm md:text-base overflow-y-scroll scrollbar">
```

We've got this utility class `scrollbar` that we use for our vertical scrollbar to make it some bit of uniform around the project. Let's use that here.

See it in action...

[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2024.09.14-10_45_16.webm](https://github.com/user-attachments/assets/dfb650cc-7920-4a56-adb6-2c37b8a0da50)

**Review by babblebey — APPROVED** [review]
This looks great @jomefavourite ❤️ 

Thank you lots for this one. I'll approve this, but I'll wait for you to address the nits in review comment to improve this, then we merge 😁

**Review by babblebey — APPROVED** [review]

**Pull request merged**